middlewares/checkDb.js

```diff
@@ -0,0 +1,32 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import models from '../models';
+
+dotenv.config();
+
+const { users } = models;
+
+export const checkAdmin = async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  const token = authHeader.split(' ')[1];
+  const decode = jwt.decode(token, process.env.SECRET);
+  const { email } = decode;
+  const foundUser = await users.findOne({
+    where: {
+      email
+    },
+    attributes: ['accessLevel']
+  });
+
+  switch (true) {
+    case foundUser === null:
+      return res.status(404).json({
+        error: 'failed to access resource'
+      });
+    case foundUser.accessLevel < 1:
+      return res.status(403).json({
+        error: 'admin/super-user authorisation required to access this resource'
```

Comment:
This error message, while very descriptive, gives away confidential details about your API to potentially rogue persons. It lets that rogue user `know` that our API supports different kinds of users, and allows them to tailor-make an attack, now that they have more detail/info. Revise the message so that it reports the error without giving away such critical detail.